### PR TITLE
script: Test_plan optimization

### PR DESCRIPTION
**Pull request by PerMac**
These commits optimize test_plan.py script workflows by preventing it from leaving blind spots and scanning horizontally when a file within a folder called "common" without yaml is modified.

### CODE REVIEW

**Comment by PerMac** [comment]
As requested, separated from https://github.com/zephyrproject-rtos/zephyr/pull/61838

**Comment by MaureenHelm** [comment]
@nashif can you take a look?

**Comment by PerMac** [comment]
rebased

File: scripts/ci/test_plan.py [hunk 1]
```diff
@@ -328,21 +349,22 @@ def find_excludes(self, skip=[]):
             ignores = filter(lambda x: not x.startswith("#"), ignores)
 
         found = set()
-        files = list(filter(lambda x: x, self.modified_files))
+        files_not_resolved = list(filter(lambda x: x not in self.resolved_files, self.modified_files))
 
         for pattern in ignores:
-            if pattern in skip:
```
Comment (gchwier) [inline scripts/ci/test_plan.py:334]: after removing this check, `skip` can be removed from method parameters

File: scripts/ci/test_plan.py [hunk 2]
```diff
@@ -328,21 +349,22 @@ def find_excludes(self, skip=[]):
...
+                found.update(fnmatch.filter(files_not_resolved, pattern))
 
         logging.debug(found)
-        logging.debug(files)
+        logging.debug(files_not_resolved)
 
-        if sorted(files) != sorted(found):
+        # Full twister run can be ordered by detecting great number of tests/boards changed
+        # or if not all modified files were resolved (corresponding scope found)
+        self.full_twister = self.full_twister or sorted(files_not_resolved) != sorted(found)
```
Comment (gchwier) [inline scripts/ci/test_plan.py:363]: if `full_twister` is set, then checking `ignore_path` can be skipped.

**Review by gchwier — APPROVED** [review]
Changes looks ok. Some minor comments (not needed to be resolved now)

**Review by nashif — APPROVED** [review]

**Pull request merged**